pod.go

```diff
@@ -389,6 +389,11 @@ func (p *Pod) Annotations(key string) (string, error) {
 	return value, nil
 }
 
+// SetOrAddAnnotation sets or adds an annotation
+func (p *Pod) SetOrAddAnnotation(key, value string) {
```

Comment:
Also, how about making this function return the previous value for `key` if it was already set. That would require 2 return values though:

```
func (p *Pod) SetOrAddAnnotation(key, value string) (string, bool) {}
```

The `bool` being set if there was a previous value (as returning `""` would be ambiguous [either the key was not set, or it was set to `""`]).